src/lib/registration/ui/addon_selection_base_dialog.rb

```diff
@@ -93,86 +101,83 @@ def details_widget
                 _("Select an extension or a module to show details here") + "</small>")))
       end
 
-      # create UI box with addon check boxes, if the number of the addons is too big
-      # the UI uses two column layout
-      # @return [Yast::Term] the main UI dialog term
-      def addons_box
-        lines = Yast::UI.TextMode ? 9 : 14
-        if @addons.size <= lines
-          content = addon_selection_items(@addons)
-        else
-          content = two_column_layout(@addons[lines..(2 * lines - 1)], @addons[0..(lines - 1)])
+      # @return [String] a Value for a RichText
+      def richtext_checkboxes(addons)
+        items = addons.map do |addon|
+          # checkbox label for an unavailable extension
+          # (%s is an extension name)
+          label = addon.available? ? addon.label : (_("%s (not available)") % addon.label)
+
+          richtext_checkbox(id:       addon_widget_id(addon),
+                            label:    label,
+                            selected: addon_selected?(addon),
+                            enabled:  addon.available?,
+                            indented: addon.depends_on)
```

Comment:
Yes, it is worse in that we send many more parameters. But the better thing is that the method does not have to care about the argument being an `Addon`.